compiler/rustc_resolve/src/diagnostics.rs

```diff
@@ -1421,14 +1419,29 @@ impl<'a, 'tcx> Resolver<'a, 'tcx> {
             "",
         );
 
+        if macro_kind == MacroKind::Bang
+            && ident.name == sym::macro_rules
+            && let source_map = self.tcx.sess.source_map()
+            && let Ok(next_source) = source_map.span_to_next_source(ident.span)
+            && next_source.starts_with("!")
+            && let next = ident.span.with_hi(ident.span.hi() + BytePos(1))
+            && let Ok(span) = source_map.span_extend_while(next, |c| c.is_whitespace() || c == '{')
```

Comment:
I still don't understand why you are trying to search for `{`.

The difference between
```
LL | macro_rules! {
   | ^^^^^^^^^^^
```
and
```
LL | macro_rules! {
   | ^^^^^^^^^^^^
```
is not even visible unless you are looking *really* carefully.
In this case it's better to just always use `ident.span`.

And if you want a span for a `name` insertion suggestion, then a zero-length span immediately after `!` would be enough.

Reply:
> I still don't understand why you are trying to search for {

It's to prevent tips from being displayed under:

```rs
macro_rules![];
macro_rules!();
```

Reply:
But why, `macro_rules! foo (() => ());` is a legal way to define a macro.

Reply:
`﻿macro_rules!` that is supposed to be an unresolved macro call and *not* a `macro_rules` item is also extremely rare, so it's not important what we do in that case, so we can optimize for the least amount of logic in the compiler.